src/ray/rpc/gcs_server/gcs_rpc_client.h

```diff
@@ -225,11 +228,20 @@ class GcsRpcClient {
     if (get_server_address_) {
       auto address = get_server_address_();
       Init(address.first, address.second, client_call_manager_);
+
+      if (Ping(address.first, address.second, 100)) {
```

Comment:
What if this ping fails? 
I think we should either retry or crash. We shouldn't proceed without re-establishing the subscriptions.

Reply:
if this ping fails, we will retry.